### PR TITLE
acts: new version 33.1.0

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds new versions of Acts through 33.1.0:
- RIP `autodiff`
- RIP `binaries`
- welcome `geomodel`

### CODE REVIEW

**Comment by wdconinc** [comment]
@spackbot fix style

**Review by HadrienG2 — APPROVED** [review]

File: var/spack/repos/builtin/packages/acts/package.py [hunk 1]
```diff
@@ -226,6 +231,7 @@ class Acts(CMakePackage, CudaPackage):
         when="@0.16:",
     )
     variant("fatras_geant4", default=False, description="Build Geant4 Fatras package")
+    variant("geomodel", default=False, description="Build GeoModel plugin", when="@33:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/acts/package.py:234]: How is this variant supposed to be used in the build?

File: var/spack/repos/builtin/packages/geomodel/package.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    depends_on("sqlite@3.7.17:")
+    depends_on("xerces-c@3.2.3:")
+
+    depends_on("geant4", when="+geomodelg4")
+    depends_on("geant4", when="+fullsimlight")
+    depends_on("hepmc3", when="+hepmc3")
+    depends_on("pythia8", when="+pythia")
+    depends_on("hdf5", when="+visualization")
+    depends_on("qt-base +gui +opengl +sql +widgets", when="+visualization")
+    depends_on("opengl", when="+visualization")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/geomodel/package.py]: Fyi. You could tighten this up a tiny bit using `with when("+visualization"):` (see the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies).

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new commits/sha256s and new package's homepage, rest LGTM except not using the new variant.

File: var/spack/repos/builtin/packages/acts/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/acts/package.py:234]: Ah, oops, forgot to add the cmake_args entry. Thanks!

Comment (wdconinc) [reply var/spack/repos/builtin/packages/acts/package.py:234]: 49836fe1356bd3fb21b8d599d784c38bcfe40018

File: var/spack/repos/builtin/packages/geomodel/package.py (hunk 2 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/geomodel/package.py]:
```suggestion
    with when("+visualization"):
        depends_on("hdf5")
        depends_on("qt-base +gui +opengl +sql +widgets")
        depends_on("opengl")
```

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**